mcs/class/System/Mono.Net.Security/MonoTlsStream.cs

```diff
@@ -157,9 +158,11 @@ public void Dispose ()
 		}
 
 		void CloseSslStream () {
-			if (sslStream != null) {
-				sslStream.Dispose ();
-				sslStream = null;
+			lock (sslStreamLock) {
```

Comment:
Just cosmetic, but how about something like this to avoid the lock:

```suggestion
			var oldStream = Interlocked.Exchange (ref sslStream, null);
			oldStream?.Dispose();
```

Reply:
Looks harder to read than a classic lock to me and since `CloseSslStream ()` is not a hot method we don't need to worry about perf of the lock :)